learning_resources/etl/oll.py

```diff
@@ -1,46 +1,186 @@
 """MITx learning_resources ETL"""
 
 import logging
+from _csv import QUOTE_MINIMAL
+from csv import DictReader
+from decimal import Decimal
+from io import StringIO
+from pathlib import Path
 
+import requests
 from django.conf import settings
 
-from learning_resources.constants import OfferedBy, PlatformType
-from learning_resources.etl.constants import ETLSource
-from learning_resources.etl.openedx import (
-    OpenEdxConfiguration,
-    openedx_extract_transform_factory,
+from learning_resources.constants import (
+    Availability,
+    OfferedBy,
+    PlatformType,
+    RunAvailability,
 )
+from learning_resources.etl.constants import ETLSource
+from learning_resources.etl.utils import generate_course_numbers_json, transform_levels
+from learning_resources.utils import get_year_and_semester
 
 log = logging.getLogger(__name__)
 
+# List of OLL course ids already ingested from OCW
+SKIP_OCW_COURSES = [
+    "OCW+18.031+2019_Spring",
+]
+
+
+def extract(sheets_id: str or None = None) -> str:
+    """
+    Extract OLL learning_resources
+
+    Args:
+        sheets_id (str): Google sheets id
+
+    Returns:
+        str: OLL learning_resources data as a csv-formatted string
+
+    """
+    if sheets_id:
+        return requests.get(
+            f"https://docs.google.com/spreadsheets/d/{sheets_id}/export?format=csv",
+            timeout=settings.REQUESTS_TIMEOUT,
+        ).content.decode("utf-8")
+    with Path.open(
+        Path(settings.BASE_DIR, "learning_resources/data/oll_metadata.csv"), "r"
+    ) as csv_file:
+        return csv_file.read()
+
+
+def transform_image(course_data: dict) -> dict:
+    """
+    Transform a course image into our normalized data structure
+
+    Args:
+        course_data (dict): course data extracted from csv/sheet
+
+    Returns:
+        dict: normalized course image data
+
+    """
+    return {
+        "url": course_data["Course Image URL Flat"],
+        "alt": course_data["title"],
+    }
+
+
+def parse_topics(course_data: dict) -> list[dict]:
+    """
+    Transform course topics into our normalized data structure
+
+    Args:
+        course_data (dict): course data
 
-def get_open_edx_config():
+    Returns:
+        dict: list of normalized course topics data
     """
-    Return the OpenEdxConfiguration for edX.
+    return [
+        # One topic name from GSheets is slightly incorrect
+        {"name": topic.replace("Educational Policy", "Education Policy")}
+        for topic in [
+            course_data["MITxO Primary Child"],
+            course_data["MITxO Secondary Child "],
```

Comment:
Is the trailing space on this field intentional? I also didnt see this column in the sheet - is this supposed to be "MITxO Primary Parent"?

Reply:
"MITxO Secondary Child " is in the online sheet (but hidden) with a trailing space in the column title, so I left it that way too in the included csv file to keep them compatible.  If you download the sheet as a csv manually ("File"->"Download"->"csv"), it will be there.

Reply:
The parent topic for each will be assigned automatically by MIT Learn during ingestion, so only the child is needed.

For example, [according to this spreadsheet](https://docs.google.com/spreadsheets/d/1Euu4ezPKeYkiTThvmxB5vFONFNG4fOULNW42F6qLuFI/edit?pli=1&gid=431773359#gid=431773359), "Machine Learning" is a child of "Data Science, Analytics & Computer Technology" while "Physics" is a child of "Math and Science", and all 4 show up for the OLL course "Mechanics: Kinematics and Dynamics".

<img width="438" alt="Screenshot 2024-09-04 at 10 39 45 AM" src="https://github.com/user-attachments/assets/0811882c-f031-4617-a8e0-7f5603934612">